packages/datadog-instrumentations/src/cucumber.js

```diff
@@ -90,7 +90,18 @@ function wrapRun (pl, isLatestVersion) {
       if (!pickleResultByFile[testSuiteFullPath]) { // first test in suite
         testSuiteStartCh.publish(testSuiteFullPath)
       }
-      testStartCh.publish({ testName: this.pickle.name, fullTestSuite: testSuiteFullPath })
+      let testSourceLine
+      try {
+        testSourceLine = this.gherkinDocument.feature.location.line
```

Comment:
If the latter then it might be more efficient to skip the try catch and write out the ugly lookup chain:

```js
const testSourceLine = this.gherkinDocument &&
  this.gherkinDocument.feature &&
  this.gherkinDocument.feature.location &&
  this.gherkinDocument.feature.location.line
```

I think this is the pattern we usually follow. It's also clearer that the intent is to look through potentially missing fields and not that a getter might throw.

That said maybe the duplicate accesses costs us performance...

I guess it's up to you. Maybe just change the catch comment to `// we can't use ?. yet` and call it a day.

Reply:
it's indeed for emulating `?.`. I'll follow your approach for consistency

Reply:
done in https://github.com/DataDog/dd-trace-js/pull/3011/commits/54d4f53f5bec962c0ea0ca3c8b9778a2eef6607e